### ISSUE TITLE
Migrate to 11ty

### ISSUE THREAD

**Issue by muffinresearch** [mozilla/extension-workshop#654]
In order to side-step the huge build times under Jekyll we are going to migrate to 11ty. The goals are to reduce build time and make this site much easier to contribute to.

There's a work in progress branch for this here: https://github.com/muffinresearch/extension-workshop/tree/migrate-to-11ty

## Core Site

- [x] Get the overall site building
- [x] SEO Plugin replacement
- [x] Parity with existing search
- [x] Fix TOC
- [x] Build tags
- [x] Fix next links
- [x] Remove the `_` from directories
- [x] Add a content directory for the markdown.
- [x] Check that drafts are not published.
  - [x] Remove superfluous `published: true` from frontmatter.
- [x] Configure 404 locally.
- [x] Use markdown code fences for all markdown syntax highlighting.
- [x] Add edit on github link #656

## Static Assets

- [x] Check page weight with released version.
- [x] Ensure, CSS and JS is using compression for prod build. 
- [x] Generate cache-busting content hash based URLs for CSS and JS and images.

## Deployment

- [x] Update / check build scripts

## Docs

- [x] Update README to reflect how this works under 11ty rather than Jekyll.

**Comment by rugk** [comment]
Just because I'm curious: Why do you do so? Do you have a more elaborate reasoning/blog post or so on that topic?

You just mentioned "to reduce build time and make this site much easier to contribute to."

**Comment by caitmuenster** [comment]
@rugk There probably isn't enough content here to justify a blog post. :) When we originally started this project, we imagined hosting this site through Github and made the decision to go with Jekyll. Later we made some decisions to move away from Github hosting but never re-evaluated whether Jekyll still made sense, and once we had the site up and running we started experiencing build times that took upwards of 13 minutes. This would occasionally time out CircleCI and generally drove all of us crazy any time we needed to check out a change. 

Based on early tests, build times are [a lot faster](https://twitter.com/muffinresearch/status/1280087200978223104) and Eleventy has some built-in features that will fix some of our SEO bugs. Plus, it should be cleaner and easier to add other features (like l10n support!) once the port is complete. 

Experience is the best teacher and I know I've definitely learned a lot from this project. :) Hopefully we'll have smoother sailing in the future.
